src/DSDPSolverInterface.jl

```diff
@@ -0,0 +1,242 @@
+    # in SOI, convention is MAX but in DSDP, convention is MIN so we reverse the sign of coef
+    _setcoefficient!(m, -coef, 0, blk, i, j)
+end
+
+function MOI.optimize!(m::DSDPSolverInstance)
+    if !isempty(m.lpdvars)
+        m.lpcone = CreateLPCone(m.dsdp)
+        LPCone.SetDataSparse(m.lpcone, m.nlpdrows, m.nconstrs+1, m.lpdvars, m.lpdrows, m.lpcoefs)
+    end
+
+    Setup(m.dsdp)
+    Solve(m.dsdp)
+
+    m.x_computed = false
+    # m.y does not contain the solution y
+    m.y_valid = false
+    m.z_computed = true
+end
+
+function MOI.get(m::DSDPSolverInstance, ::MOI.TerminationStatus)
+    status = StopReason(m.dsdp)
+    if status == DSDP_CONVERGED
+        return MOI.Success
+    elseif status == DSDP_INFEASIBLE_START
+        return MOI.OtherError
+    elseif status == DSDP_SMALL_STEPS
+        return MOI.SlowProgress
+    elseif status == DSDP_INDEFINITE_SCHUR_MATRIX
+        return MOI.NumericalError
+    elseif status == DSDP_MAX_IT
+        return MOI.IterationLimit
+    elseif status == DSDP_NUMERICAL_ERROR
+        return MOI.NumericalError
+    elseif status == DSDP_UPPERBOUND
+        return MOI.ObjectiveLimit
+    elseif status == DSDP_USER_TERMINATION
+        return MOI.Interrupted
+    elseif status == CONTINUE_ITERATING
+        return MOI.OtherError
+    else
+        error("Internal library error: status=$status")
+    end
+end
+
+MOI.canget(m::DSDPSolverInstance, ::MOI.PrimalStatus) = GetSolutionType(m.dsdp) != DSDP_PDUNKNOWN
+function MOI.get(m::DSDPSolverInstance, ::MOI.PrimalStatus)
+    compute_x(m)
+    status = GetSolutionType(m.dsdp)
+    if status == DSDP_PDUNKNOWN
+        return MOI.UnknownResultStatus
+    elseif status == DSDP_PDFEASIBLE
+        return MOI.FeasiblePoint
+    elseif status == DSDP_UNBOUNDED
+        return MOI.InfeasiblePoint
+    elseif status == DSDP_INFEASIBLE
+        return MOI.InfeasibilityCertificate
+    else
+        error("Internal library error: status=$status")
+    end
+end
+
+MOI.canget(m::DSDPSolverInstance, ::MOI.DualStatus) = GetSolutionType(m.dsdp) != DSDP_PDUNKNOWN
+function MOI.get(m::DSDPSolverInstance, ::MOI.DualStatus)
+    compute_x(m)
+    status = GetSolutionType(m.dsdp)
+    if status == DSDP_PDUNKNOWN
+        return MOI.UnknownResultStatus
+    elseif status == DSDP_PDFEASIBLE
+        return MOI.FeasiblePoint
+    elseif status == DSDP_UNBOUNDED
+        return MOI.InfeasibilityCertificate
+    elseif status == DSDP_INFEASIBLE
+        return MOI.InfeasiblePoint
+    else
+        error("Internal library error: status=$status")
+    end
+end
+
+function SOI.getprimalobjectivevalue(m::DSDPSolverInstance)
+    -GetPPObjective(m.dsdp)
+end
+function SOI.getdualobjectivevalue(m::DSDPSolverInstance)
+    -GetDDObjective(m.dsdp)
+end
+
+function compute_x(m::DSDPSolverInstance)
+    if !m.x_computed
+        ComputeX(m.dsdp)
+        m.x_computed = true
+        m.z_computed = false
```

Comment:
Is this correct?

Reply:
Yes, it is quite weird but when `ComputeX` is called, the Z vector is used (I guess to avoid new allocation) and so its value is overwritten. We need to call `ComputeS` then to recompute it.

Reply:
Haha ok